docs: Update containerized-gadgets design document (#2155)
inspektor-gadget/inspektor-gadget: docs/design/002-containerized-gadgets.md

TODO: Revisit this appproach and check if there is a way to remove the limitation of being loaded by
other projects.

#### Toppers

Toppers are gadgets that print a list of elements sorted by specific parameters, like TCP
connections by sent bytes. The gadget collects and stores the information on a hash map,and then
Inspektor Gadget automatically reads and sorts those maps according to the configuration provided by
the user:
- Interval: How often to print the map contents
- Sort by: Field within the event type to sort by

> Gadget developers can use `GADGET_STATS_MAP` to indicate the map is used to store stats.

Reviewer: ```suggestion
Gadget developers can use `GADGET_STATS_MAP` to mark the map which stores the stats.
```

```c
struct file_id {
	__u64 inode;
	__u32 dev;
	__u32 pid;
	__u32 tid;
};

struct file_stat {
	mnt_ns_id_t mntns_id;
	__u64 reads;
	__u64 rbytes;
	__u64 writes;
	__u64 wbytes;
	__u32 pid;
	__u32 tid;
	__u8 file[PATH_MAX];
	__u8 comm[TASK_COMM_LEN];
	//...
};

struct {
	__uint(type, BPF_MAP_TYPE_HASH);
	__uint(max_entries, MAX_ENTRIES);
	__type(key, struct file_id);
	__type(value, struct file_stat);
} stats SEC(".maps");

// Describes the type produced by this program.
GADGET_STATS_MAP(stats);
```

#### Snapshotters

Programs of type `iter/` are automatically loaded and attached by Inspektor Gadget, then they are
triggered one time and its output is parsed according to the BTF information provided by it.

> The `GADGET_SNAPSHOTTER` macro is available to indicate the type produced by a given snapshotter

Reviewer: ```suggestion
The `GADGET_SNAPSHOTTER` macro is available to indicate the struct type produced by a given snapshotter
```
